File: backend-api/UniYouth.Api/Shared/Enums/NotificationPriority.cs
namespace UniYouth.Api.Shared.Enums
{
    /// <summary>
    /// Mức độ ưu tiên của thông báo
    /// </summary>
    public enum NotificationPriority : byte
    {
        Normal = 0,   // Bình thường
        High = 1,     // Quan trọng
        Critical = 2 // Khẩn cấp
    }
}

File: mobile-app/lib/domain/entities/notifications/notification_priority.dart
enum NotificationPriority {
  normal,
  high,
  critical,
  unknown,
}

extension NotificationPriorityParser on NotificationPriority {
  static NotificationPriority fromApiValue(Object? rawValue) {
    if (rawValue == null) {
      return NotificationPriority.unknown;
    }

    if (rawValue is int) {
      switch (rawValue) {
        case 0:
          return NotificationPriority.normal;
        case 1:
          return NotificationPriority.high;
        case 2:
          return NotificationPriority.critical;
        default:
          return NotificationPriority.unknown;
      }
    }

    final normalized = rawValue.toString().trim().toLowerCase();
    if (normalized.isEmpty) {
      return NotificationPriority.unknown;
    }

    switch (normalized) {
      case '0':
      case 'normal':
        return NotificationPriority.normal;
      case '1':
      case 'high':
        return NotificationPriority.high;
      case '2':
      case 'critical':
        return NotificationPriority.critical;
      default:
        return NotificationPriority.unknown;
    }
  }
}

